test/mock-backend-server/src/main/java/org.wso2am.micro.gw.mockbackend/MockBackEndServer.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2am.micro.gw.mockbackend;
+
+import com.sun.net.httpserver.HttpServer;
+import java.util.logging.Level;
+import java.net.HttpURLConnection;
+import java.net.InetSocketAddress;
+import java.util.logging.Logger;
+
+/**
+ * Mock HTTP server for testing Open API tests.
+ */
+public class MockBackEndServer extends Thread {
+
+    private static final Logger logger = Logger.getLogger(MockBackEndServer.class.getName());
+    private HttpServer httpServer;
+    private String backEndServerUrl;
+    private static int backEndServerPort;
+    private static boolean retryDone = false;
+
+
```

Comment:
Shall we remove these extra lines if the file is not auto-generate?

Reply:
This is already removed